### DISCUSSION TITLE
403 Error when trying to add a site to Panopticon

### DISCUSSION THREAD

**Discussion by GJSchaller** [akeeba/panopticon#547] [Q&A (Community Support)] [answered]
I have a test site I am working with, that I am now having problems adding to Panopticon.  Here's the steps involved:

1) Main site (J 3.10) was cloned into a testing site before Panopticon was installed.
2) Panopticon was set up, and both sites were added successfully as their own URLs / Sites
3) I messed with the test site, and deleted it.  At this time, I also deleted the Test site from Panopticon.
4) I cloned the Main site back into the Testing environment
5) I removed the Panopticon connector from the Test site.
6) I upgraded the test site to J4, then J5
7) I re-added the Panopticon connector (for J4 this time)
8) I created an API Key
9) When I try to add the Test Site to Panopticon, using the URL and Key listed in the Connector, I get a 403 Forbidden result
10) I've tried:
- Removing and re-installing the connector
- Resetting the API key
- Adding the Connector to my WAF Exceptions
- Adding the Panopticon IP to my WAF Exceptions

Any chance I'm missing something?  Environment is:
Apache version: 2.4.57
PHP version: 8.2.15

Thank you for your guidance!

**Comment by nikosdion** [comment]
The first thing to check: your `.htaccess` file. Try replacing it with the contents of Joomla's `htaccess.txt` file.

The second thing to check is whether it's a `.htaccess` issue you don't understand. Start going up the directory structure of your server. Is there another `.htaccess` file there, _even if it does not belong to your test site_? The .htaccess file contents cascade across filesystem directories. This is why, for example, having a subdomain's root inside the main site's `public_html` directory is always a bad idea.

Next up, check whether you have any plugins which prevent access to Joomla's API application (the `api` folder).

Finally, check if the host is doing something stupid, like preventing access to the `api` folder of your site. And yes, I have seen that and worse.

**Comment by GJSchaller** [comment]
Thank you for the steps, here's what I've seen so far...

The site is on a host with multiple other Joomla sites.  All of them are under the same folder on the webserver:

/home/aiskon/domains/subdomain1
/home/aiskon/domains/subdomain2
/home/aiskon/domains/subdomain3

The other subdomains work fine, it's only this one site that seems to be having the issue, so that helps narrow it down (it's not an htaccess further up the chain, or a server configuration issue).

I went into the site that is having the problem, and in Admin Tools, disabled the option for "Disable Directory Listings" and rebuilt the htaccess file.  Now, when I try to add the site to Panopticon, I get an HTTP 500 error instead.  When I browse to https://testsite.aiskon.net/api/, I get: "Unable to handle request for route ``."

I'll do some more digging as time permits, but I suspect this may be an internal issue with Joomla itself?  I'll try reinstalling the core files to be safe.  The site is old - it's been around since the 1.5 days - so there's a lot of cruft in there I can't remove from the console any more, as the extensions are out of date and unsupported.

**Reply by nikosdion** [reply]
You must have some third party plugin which is doing something wrong. For example, when I visit https://testsite.aiskon.net/api/index.php/v1/panopticon/version I expect to see a "Resource not found" error message from the API. Instead of that, I see "Unable to handle request for route `v1/panopticon/version`" which **IS NOT** how the Joomla! API application works.

That's as far as I can help without having direct access to your site.

**Comment by GJSchaller** [comment, marked as answer]
I disabled everything on my site, then re-enabled it one by one until I could reproduce the issue.  It looks like it's the "System - Kunena Forum" (v6.2.3) plugin.

Thank you for your help and insight... the logic of narrowing it down is what got me to this.  Turns out it wasn't an old extension that hadn't been removed, but an update to a current one.

**Comment by GJSchaller** [comment]
After doing some more digging, it's specifically the "Kunena - Community Builder Integration" plugin for Kunena.
